examples/Web/RequestTimerMiddleware.cs

```diff
@@ -0,0 +1,49 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+
+
+namespace Web
+{
+    /// <summary>
+    ///     Middleware that times how long a web request takes to be handled,
+    ///     and logs this to Kafka.
+    /// </summary>
+    public class RequestTimerMiddleware
+    {
+        private readonly string topic;
+        private readonly KafkaDependentProducer<string, long> producer;
+        private readonly RequestDelegate next;
+
+        public RequestTimerMiddleware(RequestDelegate next, KafkaDependentProducer<string, long> producer, IConfiguration config)
+        {
+            this.next = next;
+            this.producer = producer;
+            topic = config.GetValue<string>("Kafka:RequestTimeTopic");
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            Stopwatch s = new Stopwatch();
+            try
+            {
+                s.Start();
+                await next(context);
+            }
+            finally
+            {
+                s.Stop();
+
+                // Non-blocking fire-and-forget produce call. Any errors will go unreported / will not be handled.
```

Comment:
They should at least be logged, otherwise we can just remove this produce altogether.
People will need some error indication when things are badly configured (such as topic auth).

Reply:
I think fire-and-forget is pretty appropriate here. if there are problems, there will be just be holes in the data. I don't understand how removing the produce altogether could be better (it would render this class pointless). anyway, changed to log out of band, and if people are using kafka for their logging too then they can figure it out.